src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -1020,11 +1020,12 @@ private void LoadExtensions(Assembly assembly, bool throwOnError)
         /// Gets a value indicating whether <paramref name="assembly"/> can contain extensions.
         /// </summary>
         /// <remarks>
-        /// All ML.NET product assemblies won't contain extensions.
+        /// All ML.NET product assemblies won't contain extensions besides Microsoft.ML.Samples.
         /// </remarks>
         private static bool CanContainExtensions(Assembly assembly)
         {
             if (assembly.FullName.StartsWith("Microsoft.ML.", StringComparison.Ordinal)
+                && !assembly.FullName.StartsWith("Microsoft.ML.Samples", StringComparison.Ordinal)
```

Comment:
>   && !assembly.FullName.StartsWith("Microsoft.ML.Samples", StringComparison.Ordinal) [](start = 13, length = 85)

code change detected!!! could you do this as a separate PR with proper shiproom triage? we don't want to mix code-change with documentation PRs. :)